File: Api/Companies/iCompanyPortal.Api.Companies.Client/CompanyUsersClient.cs
using iCompanyPortal.Api.Notifications.Client;
using System;
using System.Collections.Generic;
using System.Net;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace iCompanyPortal.Api.Companies.Client
{
    public class CompanyUsersClient : ICompanyUsersClient
    {
        private readonly HttpClient http;

        public CompanyUsersClient(HttpClient http)
        {
            this.http = http;
        }

        public async Task<CompanyUser> GetAsync(int companyId)
        {
            return await http.GetAsync<CompanyUser>($"/{companyId}");
        }

        public async Task<CompanyUser[]> GetAllAsync(int companyId)
        {
            return await http.GetAsync<CompanyUser[]>($"/{companyId}/all");
        }

        public async Task NotifyAsync(int companyId, NotifyRequest request)
        {
            await http.PostAsync($"/{companyId}/notify", request);
        }

        public async Task DeleteAsync(int companyId, int userId)
        {
            await http.DeleteAsync($"/{companyId}/{userId}");
        }
    }
}

File: App/icompanyportal/projects/companies-api/src/lib/services/company-users-client.ts
import { Injectable, Inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { COMPANIES_API_ENDPOINT } from './companies-api-endpoint';
import { CompanyUser } from '../models/company-user';
import { NotifyRequest } from 'notifications-api';
import { CompanyUserPermissionType } from '../models/company-user-permission-type';

@Injectable({
  providedIn: 'root'
})
export class CompanyUsersClient {
  constructor(private http: HttpClient, @Inject(COMPANIES_API_ENDPOINT) private endpoint: string) {}

  getAll(companyId: number) {
    return this.http.get<CompanyUser[]>(`${this.endpoint}/users/${companyId}/all`);
  }

  get(companyId: number) {
    return this.http.get<CompanyUser>(`${this.endpoint}/users/${companyId}`);
  }

  setPermission(companyId: number, userId: number, permissionType: CompanyUserPermissionType, isSet: boolean) {
    return this.http.put<void>(`${this.endpoint}/users/${companyId}/${userId}/${permissionType}/${isSet}`, undefined);
  }

  notify(companyId: number, request: NotifyRequest) {
    return this.http.post<void>(`${this.endpoint}/users/${companyId}/notify`, request);
  }

  delete(companyId: number, userId: number) {
    return this.http.delete<void>(`${this.endpoint}/users/${companyId}/${userId}`);
  }
}
